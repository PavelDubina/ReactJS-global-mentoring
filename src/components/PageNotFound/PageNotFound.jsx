import React from 'react'
import { Link } from 'react-router-dom'
import { Button } from '../Button/Button'
import styles from './PageNotFound.scss'

export const PageNotFound = () => (
  <div className={styles.container}>
    <h1>Page Not Found</h1>
    <p>404</p>
    <Link to="/">
      <Button styleType="back_home">GO BACK TO HOME</Button>
    </Link>
  </div>
)
