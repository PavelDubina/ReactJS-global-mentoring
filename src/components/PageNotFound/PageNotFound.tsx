import React from 'react'
import { useRouter } from 'next/router'
import { useTranslation } from 'next-i18next'
import { Button } from '../Button/Button'
import styles from '../../../styles/PageNotFound.module.scss'

export const PageNotFound: React.FC = () => {
  const { t } = useTranslation()
  const router = useRouter()
  const goHome = () => router.push('/')
  return (
    <div className={styles.container}>
      <h1>{t('404.title')}</h1>
      <p>404</p>
      <Button onClick={goHome} styleType="back_home">
        {t('404.btn')}
      </Button>
    </div>
  )
}
