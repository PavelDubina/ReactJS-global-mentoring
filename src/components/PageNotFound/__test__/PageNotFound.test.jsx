import React from 'react'
import { render } from '@testing-library/react'
import { createMemoryHistory } from 'history'
import { Router } from 'react-router-dom'
import { PageNotFound } from '../PageNotFound'

describe('PageNotFound', () => {
  const history = createMemoryHistory()
  const getComponent = () =>
    render(
      <Router history={history}>
        <PageNotFound />
      </Router>,
    )
  it('Should renders correctly', () => {
    expect(getComponent().asFragment()).toMatchSnapshot()
  })
})
