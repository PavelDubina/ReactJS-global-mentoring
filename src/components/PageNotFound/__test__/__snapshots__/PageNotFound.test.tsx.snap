// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageNotFound Should renders correctly 1`] = `
<DocumentFragment>
  <div
    class="container"
  >
    <h1>
      404.title
    </h1>
    <p>
      404
    </p>
    <button
      class="back_home"
      type="button"
    >
      404.btn
    </button>
  </div>
</DocumentFragment>
`;
